src/directions/layers.ts

```diff
@@ -5,16 +5,20 @@ import type { LayerSpecification } from "maplibre-gl";
  * {@link https://github.com/smellyshovel/maplibre-gl-directions/blob/main/src/directions/layers.ts#L3|standard
  * `MapLibreGlDirections` layers} with optionally scaled features.
  *
- * @param {number=} pointsScalingFactor A number to multiply the initial points' dimensions by
- * @param {number=} linesScalingFactor A number to multiply the initial lines' dimensions by
- * @returns {LayerSpecification[]}
+ * @param sourceName A name of the source used by the instance and layers names' prefix
+ * @param pointsScalingFactor A number to multiply the initial points' dimensions by
+ * @param linesScalingFactor A number to multiply the initial lines' dimensions by
  */
-export default function layersFactory(pointsScalingFactor = 1, linesScalingFactor = 1): LayerSpecification[] {
+export default function layersFactory(
+  sourceName = "maplibre-gl-directions",
+  pointsScalingFactor = 1,
+  linesScalingFactor = 1,
+): LayerSpecification[] {
```

Comment:
By adding a new function argument at the beginning, won't the existing examples break?

Reply:
@wipfli That's a great point! It did actually not only break the existing examples, but the backwards-compatibility as well! And, as you could see, the build passed with no issues because, as it turned out, type checking was disabled for the Demo-project. Now it all should be fixed and the new param is moved to the end so that if it's `undefined` in existing projects then nothing's broken (thanks to the default value). Thank you for noticing this!

Reply:
That's why we need code-review I believe. Sometimes obvious things are just get lost in sight.

Reply:
Nice :)